### ISSUE TITLE
Bug: [@typescript-eslint/no-unused-vars"] incorrect usage detection 

### ISSUE THREAD

**Issue by iselo** [typescript-eslint/typescript-eslint#7342]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&fileType=.ts&code=MYGwhgzhAEDKD2BbApgYXFaBvAUD60ADgE4CWAbmAC7LQpUAW8AJgIwAUhYxY9yxrAFzQwAOwCeAGiLdeyGsQBMwseICU2fAWjB4oiPBDIAdCHgBzTrL4C1WgL54CJCtVr0mzRVZ42hIiWkuX3l%2BZQD1TW0dPQMjUwsfOQVFOwJHLRdKGjp5TwBmThVAmRCFf1Ug61ClYsjcaN19QxMzS2DksLToDIzRZAB3OCQ0DAh2NSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaWXWZRAE0IDcBDaVDSfAMxcgBpxtIABAFwCeABxQBjaAEth-QiniTc-APQFipclToN0URNGj5okbgF8QpoA&tsconfig=&tokens=false

### Repro Code

```typescript
class SomeClass {

  private method1(parameter1: any, parameter2: any) {
    console.log(parameter1)
  }

  private method2(parameter1: any, parameter2: any) {
    console.log(parameter2)
  }

  private method3(p: any, parameter1: any, parameter2: any) {
    console.log(parameter2)
  }

}

new SomeClass();
```

### ESLint Config

```javascript
{
  "rules": {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

### Expected Result

'parameter2' is defined but never used. 3:36 - 3:51
'parameter1' is defined but never used. 7:19 - 7:34
'p' is defined but never used. 11:19 - 11:25
'parameter1' is defined but never used. 11:27 - 11:42

### Actual Result

'parameter2' is defined but never used. 3:36 - 3:51

**Comment by bradzacher** [comment]
https://eslint.org/docs/latest/rules/no-unused-vars#args

The default option is `after-used`

**Comment by iselo** [comment]
.eslintrc.json
~~~json

  "rules": {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": [
      "warn",
      {
        "args": "all"
      }
    ] 
  }
~~~
*Note:  ESLint throws 'no-unused-vars' for TypeScript interface method arguments.*
